### PR TITLE
CALS1-5658-Set Default Search Results Value to 10 / Refactor facility search

### PR DESCRIPTION
**Pull request by sunilb0575**
#### Which User story does this relate to?
- https://osi-cwds.atlassian.net/browse/CALS-5658

#### Brief feature description
-Change search results to 10 
- Refactoring, moving states from child component to index 

#### Does this PR block/effect other stories?
- NO

#### Do any commands needs to be run in order to run the project with this PR?
- NO

#### Code coverage percent diff as a result of this PR?
- Branches- 70

### CODE REVIEW

**Comment by Nasir2712** [comment]
jenkins, retest this please

File: app/javascript/search/search_Data.js [hunk 1]
```diff
@@ -4,51 +4,26 @@ import {resultsPerPage} from './common/commonUtils'
 import {dictionaryNilSelectValue, floatToNextInt} from 'helpers/commonHelper.jsx'
```
Comment (ratneshraval) [inline app/javascript/search/search_Data.js]: propTypes and defaultProps for this component.

File: app/javascript/search/search_Data.js [hunk 2]
```diff
@@ -4,51 +4,26 @@ import {resultsPerPage} from './common/commonUtils'
...
-    if (this.state.searchData[3] && this.state.searchData[3].length > 1) {
-      searchFacilityName = (<p>Facility Name:
-      <span>{this.state.searchData[3]}</span>
-      <span onClick={this.removeCriteria.bind(this, this.state.searchData[3])} alt='cross-icon' className='cross-icon' />
-      </p>)
+    if (this.props.inputData.facilityNameValue) {
+      searchFacilityName = (
+        <p>Facility Name:
+        <span>{this.props.inputData.facilityNameValue}</span>
+        <span onClick={() => { this.props.handleInputChange('facilityNameValue', ''); this.props.searchApiCall(this.props.fromValue, this.props.sizeValue) }} alt='cross-icon' className='cross-icon' />
```
Comment (ratneshraval) [inline app/javascript/search/search_Data.js]: I see what you're doing here. Any way to format it little better so we can see that 2 lines are executing?

**Comment by ratneshraval** [comment]
@sunilb0575 @Nasir2712  Some overall comments 
1. nextPage and previousPage status can be calculated from pagenumber and total. Those do not need to be in state.
2. Also when you convert those to constant, they really belong in `search_Data.js`, there isn't really a use for them in `search.js`
3. I don't think you don't need 2 separate functions `backToPreviousPage` and `changeToNextPage`. Use a single function `changePage` and pass the appropriate page number when you call it

File: app/javascript/search/search_Data.js [hunk 3]
```diff
@@ -4,51 +4,29 @@ import {resultsPerPage} from './common/commonUtils'
...
-        }
-      )
-    }
-    this.props.searchApiCall(this.state.searchData.toString(), 0, this.props.sizeValue)
-  }
-
   render () {
     const searchCount = this.props.totalNoOfFacilities
     const noOfPages = floatToNextInt(searchCount, this.props.sizeValue)
+    const fromValue = this.props.sizeValue * (this.props.pageNumber - 1)
```
Comment (ratneshraval) [inline app/javascript/search/search_Data.js]: You're calculating fromValue here and on `search.js`, can you move it to a function?

File: app/javascript/search/search_Data.js [hunk 4]
```diff
@@ -4,51 +4,29 @@ import {resultsPerPage} from './common/commonUtils'
...
+    let disableNext = fromValue + this.props.sizeValue >= searchCount || searchCount < 5
```
Comment (ratneshraval) [inline app/javascript/search/search_Data.js]: Do you need `|| searchCount < 5`?

File: app/javascript/search/search_Data.js [hunk 5]
```diff
@@ -4,51 +4,29 @@ import {resultsPerPage} from './common/commonUtils'
...
+    let disablePrevious = fromValue - this.props.sizeValue < 0 || fromValue === 0
```
Comment (ratneshraval) [inline app/javascript/search/search_Data.js]: You also don't need `|| fromValue === 0`

File: app/javascript/search/search_input.js [hunk 6]
```diff
@@ -43,14 +16,14 @@ export default class SearchInput extends React.Component {
     )
     return (
       <div className='search_input col-xs-12 col-sm-12 col-md-12 col-lg-12'>
-        <form onSubmit={this.sendSearchInput.bind(this)}>
+        <form onSubmit={(event) => { event.preventDefault(); this.props.searchApiCall(this.props.fromValue, this.props.sizeValue) }}>
```
Comment (ratneshraval) [inline app/javascript/search/search_input.js]: You aren't using event here. You can rewrite this as 
```
{this.props.searchApiCall(this.props.fromValue, this.props.sizeValue)}
```

**Comment by sunilb0575** [comment]
jenkins, retest this please

**Comment by Nasir2712** [comment]
@sunilb0575  can you update nesh's comments on with the update saying if its resolved or not.
Thanks

**Comment by sunilb0575** [comment]
@ratneshraval  @Nasir2712  Addressed above mentioned overall comments 1,2,3.

**Pull request merged**